### PR TITLE
2-Apply example ynh

### PR DESCRIPTION
**Pull request by yalh76**
## Problem
- *Apply example_ynh*

## Solution
- *Applying example_ynh*

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
- [ ] **Code review**
- [ ] **Approval (LGTM)**  

- **CI succeeded** : 
Capture

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,9 +26,40 @@ Don't hesitate to give a hand if you wish, I assume only the nginx file needs im
 * to be added:
   * [ ] everything
 
+## Screenshots
+
+![](https://camo.githubusercontent.com/c1c2e74057dcff57e103fcbb3239840802fcf752/68747470733a2f2f706978656c6665642e6e7963332e63646e2e6469676974616c6f6365616e7370616365732e636f6d2f6d656469612f53637265656e25323053686f74253230323031392d30322d30352532306174253230362e33342e3539253230504d2e706e67)
+
+## Documentation
+
+ * Official documentation: https://docs.pixelfed.org/master/
```
Comment (Jibec) [inline README.md]: please move this to Links

File: manifest.json [hunk 2]
```diff
@@ -1,52 +1,55 @@
...
-        "default": "0"
-      }
-    ]
-  }
+    "name": "Pixelfed",
+    "id": "pixelfed",
+    "packaging_format": 1,
+    "description": {
+        "en": "ActivityPub Federated Image Sharing",
+        "fr": "Partage d'image fédéré avec ActivityPub."
```
Comment (Jibec) [inline manifest.json]: No translation in this file please, I'm against the downstream translation.
This is a difficult project, please see the Weblate pull requests from https://github.com/YunoHost/apps/pulls

File: manifest.json [hunk 3]
```diff
@@ -1,52 +1,55 @@
...
+    },
+    "version": "0.0.0~ynh2",
```
Comment (Jibec) [inline manifest.json]: please change the package version, at least the ynh version

File: scripts/change_url [hunk 4]
```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
```
Comment (Jibec) [inline scripts/change_url]: I'm unsure this is a feature that exists upstream, If this isn't the case, please delete this file

File: scripts/remove [hunk 5]
```diff
@@ -13,47 +13,53 @@ source /usr/share/yunohost/helpers
 #=================================================
 # LOAD SETTINGS
 #=================================================
+ynh_print_info "Loading installation settings..."
 
 app=$YNH_APP_INSTANCE_NAME
 
-domain=$(ynh_app_setting_get "$app" domain)
-db_name=$(ynh_app_setting_get "$app" db_name)
+domain=$(ynh_app_setting_get $app domain)
```
Comment (Jibec) [inline scripts/remove]: Please don't remove double quotes. They often are useless for experts, but not using it all the time makes it safer.
You can save time by using https://www.shellcheck.net to do it for you.

File: scripts/restore [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+#=================================================
+# REINSTALL DEPENDENCIES
+#=================================================
+ynh_print_info "Reinstalling dependencies..."
+
+# Define and install dependencies
+ynh_install_app_dependencies $pkg_dependencies
+
+#=================================================
+# RESTORE THE MYSQL DATABASE
```
Comment (Jibec) [inline scripts/restore]: It will require to use postgresql helpers in the future

**Comment by Jibec** [comment]
please note it may be easier for both you and me to finish the work here, then go to your branch 3-upgrade and run `git merge 2-example_ynh` and open a new PR

File: README.md (hunk 1 above)
Comment (yalh76) [reply README.md]: Done :)

File: manifest.json (hunk 2 above)
Comment (yalh76) [reply manifest.json]: all translations removed

File: manifest.json (hunk 3 above)
Comment (yalh76) [reply manifest.json]: Change made. I didn't changed it because it was just code reorganisation

File: scripts/change_url (hunk 4 above)
Comment (yalh76) [reply scripts/change_url]: Well, even if the upstream app hasn't a change_url functionnality, website can be moved from one domain to another one

File: scripts/remove (hunk 5 above)
Comment (yalh76) [reply scripts/remove]: All double quoted

File: scripts/restore (hunk 6 above)
Comment (yalh76) [reply scripts/restore]: Yes, it's planned :)

**Comment by yalh76** [comment]
> 
> 
> please note it may be easier for both you and me to finish the work here, then go to your branch 3-upgrade and run `git merge 2-example_ynh` and open a new PR

After other PR on other packages... example: https://github.com/YunoHost-Apps/z-push_ynh/pull/33

I'm trying to make review easier without too many changes in each PR.
The plan is:

1. [1-package_linter](https://github.com/yalh76/pixelfed_ynh/tree/1-package_linter): just fixing package linter, no functionnality changes
2. [2-example_ynh](https://github.com/yalh76/pixelfed_ynh/tree/2-example_ynh): just applying example_ynh, no functionnality changes
3. [3-upgrade](https://github.com/yalh76/pixelfed_ynh/tree/3-upgrade): upgrade the upstream because I wasn't able to make the actual upstream version works and using new helpers
4. implement backup/restore/upgrade/change_url

So if you prefer, we can wait step 4 and have a full PR or maybe you can create a testing branch to let me PR to that branch before a full PR from testing to the master branch.

As you wish.

PS: I'm more a beginner...

File: scripts/change_url (hunk 4 above)
Comment (Jibec) [reply scripts/change_url]: in theory yes, but it's far from being always true, sometimes it's just not possible.

Here for example, with federation using the domain name, how do you expect to change the domain name? I'm pretty sure it won't work :p

**Comment by Jibec** [comment]
thanks for your answer, I'm not an expert with git and github either, don't worry. My issue is that I have the same commits in multiple PR, making the diff harder to read globally. But I'm unsure how to prevent this.

What I wish is one feature per PR. The more little the PR is, the easier is my testing and review. But this may mean more work for you. I also tried to submit patches to yunohost apps, and to be honnest, I most of the time failed to do it ;)

One PR for install/remove, one for upgrade, one for backup/restore, etc.

File: scripts/change_url (hunk 4 above)
Comment (yalh76) [reply scripts/change_url]: I agree with you, the federation part will break if you change the domain name. But even if the federation breaks someone may want to change the name of the domain ^^

**Pull request merged**